package.json

```diff
@@ -18,28 +18,50 @@
     "scripts": {
         "dev": "rollup -m -c",
         "build": "rimraf build && rollup -m -c --environment NODE_ENV:production",
-        "test": "mocha"
+        "postbuild": "echo '{\"type\": \"commonjs\"}'> build/cjs/package.json",
+        "test": "npm run test:node && npm run test:browser",
+        "test:node": "NODE_OPTIONS='--experimental-vm-modules --no-warnings' jest --coverage",
+        "test:browser": "NODE_OPTIONS='--experimental-vm-modules --no-warnings' jest --coverage --testEnvironment=jsdom"
     },
-    "author": "Jose F. Romaniello <jfromaniello@gmail.com> (http://joseoncode.com)",
+    "author": "Jose F. Romaniello <jfromaniello@gmail.com>",
     "contributors": [
         "Sam Bellen <sam.bellen@auth0.com>"
     ],
     "license": "MIT",
     "devDependencies": {
+        "@babel/core": "^7.22.7",
+        "@babel/preset-env": "^7.22.7",
         "@rollup/plugin-commonjs": "^21.0.2",
         "@rollup/plugin-node-resolve": "^13.1.3",
-        "expect.js": "^0.3.1",
-        "mocha": "^9.2.1",
+        "@types/jest": "^29.5.3",
+        "babel-jest": "^29.6.0",
+        "jest": "^29.6.0",
+        "jest-environment-jsdom": "^29.6.0",
+        "jest-junit": "^16.0.0",
         "rimraf": "^3.0.2",
         "rollup": "^2.70.0",
         "rollup-plugin-livereload": "^2.0.5",
         "rollup-plugin-serve": "^1.1.0",
         "rollup-plugin-sourcemaps": "^0.6.3",
         "rollup-plugin-terser": "^7.0.2",
+        "rollup-plugin-typescript2": "^0.35.0",
+        "ts-jest": "^29.1.1",
+        "typescript": "^5.1.6",
         "uglify-js": "^3.15.2"
     },
     "files": [
         "build/",
         "index.d.ts"
-    ]
+    ],
+    "engines": {
+        "node": ">=16"
+    },
+    "type": "module",
```

Comment:
I could be wrong about this, but since the type is now 'module', might this cause issues with bundlers that now expect that the `main` field is in ESM, but are getting CommonJS instead?

Reply:
Ohhh, yeah. It is a massive PITA. I have tried introducing it in a minor version of Leaflet, but (build) tooling is so diverse in how it handles this that we decided to roll it back until the next major version.

Reply:
See also ["Dual Package Hazard"](https://github.com/GeoffreyBooth/dual-package-hazard).

Reply:
Based on what we understand, tooling understanding type, will understand exports. Those that don't, might want cjs.